cmd/server/main.go

```diff
@@ -2,26 +2,82 @@ package main
 
 import (
 	"encoding/json"
+	"flag"
 	"fmt"
 	"io"
 	"log"
+	"net"
 	"net/http"
+	"os"
 	"time"
 
+	"github.com/joho/godotenv"
+
 	"github.com/benchttp/engine/internal/configparse"
 	"github.com/benchttp/engine/runner"
 )
 
-const (
-	port = "8080"
+const readySignal = "READY"
+
+var (
+	// stdout is used to send messages to the parent process.
+	// Use stdout to send info and debug level messages.
+	stdout = log.New(os.Stdout, "", 0)
+	// stderr is used to send messages to the parent process.
+	// Use stderr to send error level messages or to notify
+	// any action which are followed by os.Exit.
+	stderr = log.New(os.Stderr, "", log.LstdFlags)
 )
 
 func main() {
-	addr := ":" + port
-	fmt.Println("http://localhost" + addr)
+	useAnyPort := flag.Bool("any-port", true, "use any available port allocated by the os")
+	flag.Parse()
+
+	var p string
+	if *useAnyPort {
+		p = "0"
+	} else {
+		err := godotenv.Load("./.env.development")
+		if err != nil {
+			stderr.Fatalf("could not load .env file: %s", err.Error())
+		}
+		p = os.Getenv("SERVER_PORT")
+	}
+
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:"+p)
+	if err != nil {
+		stderr.Fatal(err)
+	}
+
+	listener, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		stderr.Fatal(err)
+	}
+
+	port := listener.Addr().(*net.TCPAddr).Port
+
+	s := &http.Server{
+		Addr:    ":" + fmt.Sprint(port),
+		Handler: http.HandlerFunc(handleStream),
+		// No timeout because the end user runs both the client and
+		// the server on their machine, which makes it irrelevant.
+		ReadHeaderTimeout: 0,
+	}
+
+	closeChan := make(chan error, 1)
+
+	go func() {
+		closeChan <- s.Serve(listener)
+	}()
+
+	// Notify server is ready.
+	stdout.Println(readyString(port))
+
+	stderr.Fatal(<-closeChan)
```

Comment:
I think this is equivalent to:
```suggestion
	// Notify server is ready.
	stdout.Println(readyString(port))
	stderr.Fatal(s.Serve(listener))
```